test/function/facility.cpp

```diff
@@ -116,4 +121,78 @@ namespace {
     EXPECT_EQ( result, -1 );
   }
 
+
+TEST(GetFacilityString, InvalidFacility) {
```

Comment:
For all of these tests, use `EXPECT_ERROR_ID_EQ` to also check that the error code is appropriate. You can find examples of this in most of the other test suites.